src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,485 @@
+package duke;
+
+import duke.storage.Storage;
+
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import duke.ui.Ui;
+
+import duke.dukeException.DukeException;
+
+import duke.parser.Parser;
+
+public class Duke {
+    public static String home = System.getProperty("user.home");
+    public static String filePath = home + "/Desktop/duke.txt";
+    private Storage storeFile;
+    private TaskList tasks;
+    private Ui ui;
+    private boolean isGui = false;
+    private boolean hasExit = false;
+    private String userInput;
+
+    /**
+     * Initialize Duke, UI and load saved file into a TaskList object. If save file is empty, a new
+     * TaskList object will be created.
+     *
+     * @param filePath hard-coded File Path of the saved file
+     */
+    public Duke(String filePath) {
+        this.ui = new Ui();
+        this.storeFile = new Storage(filePath);
+        try {
+            tasks = new TaskList(storeFile.loadFile());
+        } catch (DukeException e) {
+            ui.showErrorMsg(e);
+            tasks = new TaskList();
+        }
+    }
+
+    public Duke() {
+        this(filePath);
+        isGui = true;
+    }
+
+    /**
+     * Main execution of Duke Chat bot for CLI
+     */
+    public void run() {
+        ui.printMsg(ui.startupMsg());
+        String[] userInputArray;
+        boolean validInput;
+
+        while (true) {
+            userInput = ui.readInput(isGui);
+            if (userInput.equals("bye")) {
+                break;
+            }
+            switch (userInput) {
+            case "list":
```

Comment:
i think you can abstract all the methods out because your line of code is slightly long? >30 lines